### PR TITLE
Add code to manage LDAP domain when search the computer user in GLPI DB

### PR DESCRIPTION
**Pull request by ddurieux**
Related to https://github.com/glpi-project/glpi/pull/3585

### CODE REVIEW

File: inc/formatconvert.class.php [hunk 1]
```diff
@@ -1286,9 +1286,19 @@ static function computerInventoryTransformation($array) {
             }
             if ($cnt == 0) {
                if (isset($array_tmp['login'])) {
+                  // Search on domain
+                  $where_add = '';
+                  if (isset($array_tmp['domain'])
+                          && !empty($array_tmp['domain'])) {
+                     foreach ($DB->request('glpi_authldaps',
```
Comment (trasher) [inline inc/formatconvert.class.php]: Do not call methods in a loop. Store call in a variable, and use it to loop.

File: inc/formatconvert.class.php [hunk 2]
```diff
@@ -1286,9 +1286,19 @@ static function computerInventoryTransformation($array) {
...
+                         ['WHERE'  => "`inventory_domain` = '".$array_tmp['domain']."'"]) as $data_LDAP) {
+                        $where_add = " AND `authtype`='".Auth::LDAP."' AND `auths_id`='".$data_LDAP['id']."' ";
+                     }
+                  }
                   $query = "SELECT `id`
```
Comment (trasher) [inline inc/formatconvert.class.php]: This one must be switched to dbiterator ;)

**Comment by trasher** [comment]
Maybe:
```diff
diff --git a/inc/formatconvert.class.php b/inc/formatconvert.class.php
index 043ae9c2e..0e853f725 100644
--- a/inc/formatconvert.class.php
+++ b/inc/formatconvert.class.php
@@ -1292,19 +1292,28 @@ class PluginFusioninventoryFormatconvert {
                           && !empty($array_tmp['domain'])) {
                      $ldaps = $DB->request('glpi_authldaps',
                            ['WHERE'  => "`inventory_domain` = '".$array_tmp['domain']."'"]
-                     );
+                        );
+                     $ldaps_ids = [];
+                     $where_add = [];
                      foreach ($ldaps as $data_LDAP) {
-                        $where_add = " AND `authtype`='".Auth::LDAP."' AND `auths_id`='".$data_LDAP['id']."' ";
+                        $ldaps_ids = $data_LDAP['id'];
+                     }
+                     if (count($ldaps_ids)) {
+                        $where_add['authtype'] = Auth::LDAP;
+                        $where_add['auths_id'] = $ldaps_ids;
                      }
                   }
-                  $query = "SELECT `id`
-                            FROM `glpi_users`
-                            WHERE `name` = '" . $array_tmp['login'] . "'
-                            ".$where_add."
-                            LIMIT 1";
+                  $iterator = $DB->request([
+                     'SELECT' => ['id'],
+                     'FROM'   => 'glpi_users',
+                     'WHERE'  => [
+                        'name'   => $array_tmp['login']
+                     ] + $where_add,
+                     'LIMIT'  => 1
+                  ]);
                   $result = $DB->query($query);
-                  if ($DB->numrows($result) == 1) {
-                     $a_inventory['Computer']['users_id'] = $DB->result($result, 0, 0);
+                  if ($row = $iterator>next()) {
+                     $a_inventory['Computer']['users_id'] = $row['id'];
                   }
                }
             }
```

**Comment by trasher** [comment]
As said on irc, `s/>next/->next/`

**Comment by ddurieux** [comment]
Thanks ;)

**Pull request merged**